frontend/lib/src/components/widgets/DataFrame/columns/LinkColumn.test.ts

```diff
@@ -149,4 +155,67 @@
     const cell = mockColumn.getCell("test", true)
     expect(isErrorCell(cell)).toEqual(false)
   })
+
+  it("sets the href and displayText values correctly", () => {
+    const mockColumn = LinkColumn({
+      ...MOCK_LINK_COLUMN_PROPS,
+      columnTypeOptions: { display_text: "Click me" },
+    })
+
+    const cell = mockColumn.getCell("https://streamlit.io", true) as LinkCell
+
+    const cellValue = mockColumn.getCellValue(cell)
+    expect(cellValue).toBe("https://streamlit.io")
+    expect(cell.data.displayText).toBe("Click me")
+  })
+
+  it("sets displayed value to be the href when displayText is empty", () => {
+    const mockColumn = LinkColumn({
+      ...MOCK_LINK_COLUMN_PROPS,
+      columnTypeOptions: { display_text: undefined },
+    })
+
+    const cell = mockColumn.getCell("https://streamlit.io", true) as LinkCell
+
+    expect(cell.data.displayText).toBe("https://streamlit.io")
+  })
+
+  it("sets displayed value to be displayText when displayText is defined and not a regexp", () => {
+    const mockColumn = LinkColumn({
+      ...MOCK_LINK_COLUMN_PROPS,
+      columnTypeOptions: { display_text: "streamlit" },
+    })
+
+    const cell = mockColumn.getCell("https://streamlit.io", true) as LinkCell
+
+    expect(cell.data.displayText).toBe("streamlit")
+  })
+
+  it("sets displayed value as the applied regex to the href when displayText is a regex", () => {
+    const mockColumn = LinkColumn({
+      ...MOCK_LINK_COLUMN_PROPS,
+      columnTypeOptions: { display_text: "https://(.*?).streamlit.app" },
+    })
+
+    const cell = mockColumn.getCell(
+      "https://roadmap.streamlit.app",
+      true
+    ) as LinkCell
+
+    expect(cell.data.displayText).toBe("roadmap")
+  })
+
+  it("sets displayed value as the href, when displayText is a regex but there is no match", () => {
+    const mockColumn = LinkColumn({
+      ...MOCK_LINK_COLUMN_PROPS,
+      columnTypeOptions: { display_text: "https://(.*?)/.google.com" },
```

Comment:
Theoretically we can dismiss this since it is only part of a test. But I think you need to escape the `.` via :
`https://(.*?)\.google.com` to not run into this warning.

Reply:
Shoot, my mistake 👍